From 5cf13e21c6ee812953a6664ab33041bda06017b8 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 31 Jul 2024 17:30:58 +0200
Subject: [PATCH 1122/1695] plat-stm32mp1: add internal RAMs memory layout in
 platform config header

Add STM32MP13/15 internal RAM layout in the platform config header
file.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 442c1eff8d10b2e6c9ac8c9adb648d8062af2ed1 plat-stm32mp1: add internal RAMs memory layout in platform config header]
Change-Id: Ia80c98790c1bd35a1cefe9cf1b15ebd7af56dc30
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/396252
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp1/platform_config.h | 31 +++++++++++++------
 1 file changed, 21 insertions(+), 10 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/platform_config.h b/core/arch/arm/plat-stm32mp1/platform_config.h
index 920dbe218..483d32a29 100644
--- a/core/arch/arm/plat-stm32mp1/platform_config.h
+++ b/core/arch/arm/plat-stm32mp1/platform_config.h
@@ -100,7 +100,6 @@
 #define PWR_BASE			0x50001000
 #define QSPI_BASE			0x58003000
 #define RCC_BASE			0x50000000
-#define RETRAM_BASE			0x38000000ul
 #ifdef CFG_STM32MP13
 #define RNG1_BASE			0x54004000
 #endif
@@ -110,10 +109,6 @@
 #define RTC_BASE			0x5c004000
 #ifdef CFG_STM32MP15
 #define SPI6_BASE			0x5c001000
-#define SRAM1_BASE			0x30000000
-#define SRAM2_BASE			0x30020000
-#define SRAM3_BASE			0x30040000
-#define SRAM4_BASE			0x30050000
 #endif
 #define STGEN_BASE			0x5c008000
 #define SYSCFG_BASE			0x50020000
@@ -315,14 +310,30 @@
 #endif
 
 #ifdef CFG_STM32MP15
-#define SRAM1_SIZE			0x20000
-#define SRAM2_SIZE			0x20000
-#define SRAM3_SIZE			0x10000
-#define SRAM4_SIZE			0x10000
-#endif
+/* SRAM layout*/
+#define SRAM1_BASE			U(0x30000000)
+#define SRAM1_SIZE			U(0x20000)
+#define SRAM2_BASE			U(0x30020000)
+#define SRAM2_SIZE			U(0x20000)
+#define SRAM3_BASE			U(0x30040000)
+#define SRAM3_SIZE			U(0x10000)
+#define SRAM4_BASE			U(0x30050000)
+#define SRAM4_SIZE			U(0x10000)
 
 /* RETRAM layout */
+#define RETRAM_BASE			U(0x38000000)
 #define RETRAM_SIZE			U(0x10000)
+#endif
+
+#ifdef CFG_STM32MP13
+/* SRAM layout*/
+#define SRAM1_BASE			U(0x30000000)
+#define SRAM1_SIZE			U(0x4000)
+#define SRAM2_BASE			U(0x30004000)
+#define SRAM2_SIZE			U(0x2000)
+#define SRAM3_BASE			U(0x30006000)
+#define SRAM3_SIZE			U(0x2000)
+#endif
 
 /* MCUSRAM layout */
 #define MCUSRAM_SIZE			U(0x60000)
-- 
2.47.2

